json_serialization/parser.nim

```diff
@@ -456,14 +456,33 @@ proc parseJsonNode(r: var JsonReader): JsonNode =
   of JsonValueKind.String:
     result = JsonNode(kind: JString, str: r.parseString())
   of JsonValueKind.Number:
-    var val: JsonNumber[uint64]
-    r.lex.scanNumber(val)
-    r.checkError
-    if val.isFloat:
-      result = JsonNode(kind: JFloat, fnum: r.toFloat(val, typeof(result.fnum)))
+    if JsonReaderFlag.legacyJsonNodeNumber in r.lex.flags:
```

Comment:
not sure we want to burden the implementation with this option, ie it's just bumping compatibility to what has been shipped in `std/` for a good while now.